apps/web/src/components/main/UseFilterRouter.ts

```diff
@@ -0,0 +1,36 @@
+import { usePathname, useRouter, useSearchParams } from 'next/navigation'
+import { useCallback } from 'react'
+
+import { SORT_KEYS } from './SortFilter'
+
+export type FeedFilterQueryParams = {
+  category?: number
+  org?: number
+  sort?: SORT_KEYS
+}
+
+export default function useFilterRouter() {
+  const router = useRouter()
+  const pathname = usePathname()
+  const searchParams = useSearchParams()
+
+  const moveToParams = useCallback(
+    (params: FeedFilterQueryParams) => {
+      const current = new URLSearchParams(Array.from(searchParams.entries()))
+      Object.entries(params).forEach(([key, value]) => {
+        current.set(key, value as string)
+      })
+
+      const newQueryString = current.toString()
+      const hasQueryString = newQueryString.length > 0
+      const url = hasQueryString ? `${pathname}?${newQueryString}` : pathname
+
+      router.push(url)
+    },
+    [router, pathname, searchParams],
+  )
+
+  return {
+    moveToParams,
+  }
+}
```

Comment:
이런 방법은 어떤가용? `useFilterRouter`를 사용하는 컴포넌트 내부에서 `useSearchParams`가 중복적으로 사용되더라구용! 
```suggestion

    const cateegory = useMemo(() => {
    return as<number>(searchParams.get('category') || 0)
  }, [searchParams])

  const organization = useMemo(() => {
    return as<number>(searchParams.get('organization') || 0)
  }, [searchParams])

  const sort = useMemo(() => {
    return as<number>(searchParams.get('sort') || 0)
  }, [searchParams])

  return {
    moveToParams,
    cateegory,
    organization,
    sort,
  }
}
```

Reply:
오 좋아유
백엔드쪽에서 어떻게 줄지 몰겟어서 일단 임시로 작동하게만 두었는데, 지금상태로는 저기로 옮기는게 좋겟네유👍